# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/PHttpTransaction.ipdl
# Commit: 8e7c97a37073
# Full Hash: 8e7c97a37073a14ba832e2f5c27f2c11b7787ef5
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2023-03-15 16:26:49
# Regressor Bug: 1805372
# File Overlap Count: 1
# Description:
#   Bug 1805372 - Pass effectiveTRRMode to nsHttpChannel from socketTransport/DnsAndConnectSocket r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164539
# ==============================================================================

diff -r 5abfd5ad2e35 -r 8e7c97a37073 netwerk/protocol/http/PHttpTransaction.ipdl
--- a/netwerk/protocol/http/PHttpTransaction.ipdl	Wed Mar 15 12:41:26 2023 +0000
+++ b/netwerk/protocol/http/PHttpTransaction.ipdl	Wed Mar 15 13:00:46 2023 +0000
@@ -20,6 +20,8 @@
 using mozilla::net::NetAddr from "mozilla/net/DNS.h";
 using mozilla::net::ClassOfService from "mozilla/net/ClassOfService.h";
 [RefCounted] using class nsITransportSecurityInfo from "nsITransportSecurityInfo.h";
+using mozilla::net::TRRSkippedReason from "nsITRRSkipReason.h";
+using nsIRequest::TRRMode from "nsIRequest.h";
 
 namespace mozilla {
 namespace net {
@@ -33,6 +35,8 @@
   NetAddr selfAddr;
   NetAddr peerAddr;
   bool resolvedByTRR;
+  TRRMode mode;
+  TRRSkippedReason trrSkipReason;
   bool echConfigUsed;
 };
 
@@ -52,7 +56,9 @@
                        bool                       dataToChildProcess,
                        bool                       restarted,
                        uint32_t                   HTTPSSVCReceivedStage,
-                       bool                       supportsHttp3);
+                       bool                       supportsHttp3,
+                       TRRMode trrMode,
+                       TRRSkippedReason trrSkipReason);
   async OnTransportStatus(nsresult status,
                           int64_t progress,
                           int64_t progressMax,